app/views/api_views.py

```diff
@@ -20,6 +20,34 @@ def autocomplete_name(request):
         return HttpResponse('', content_type="application/json")
 
 
+@login_required(login_url='/login')
+@require_GET
+def donor_names(request):
+    try:
+        donor_names = [donor.donor_name for donor in Donor.objects.all()]
+        return JsonResponse({'donorNames': donor_names},
+                            content_type="application/json")
+    except Exception as e:
+        return JsonResponse(e)
```

Comment:
you will prob want to return with a status=500 or something as this will be resolved as 200 in ajax resolver if not